src/ppl/src/IOPlacer.cpp

```diff
@@ -1293,6 +1635,59 @@ bool IOPlacer::assignPinsToSections(int assigned_pins_count)
   return false;
 }
 
+bool IOPlacer::assignPinsToSectionsPolygon(int assigned_pins_count)
+{
+  Netlist* net = netlist_.get();
+  std::vector<Section>& sections = sections_;
+
+  createSectionsPolygon();
```

Comment:
This call is the only thing different from `assignPinsToSections`. Merge them into a single function.